configure.ac

```diff
@@ -867,40 +867,38 @@ AC_ARG_ENABLE(delay-pools,
 dnl disable generic/common adaptation support by default
 squid_opt_use_adaptation=no
 
-squid_opt_use_esi=yes
-AH_TEMPLATE([USE_SQUID_ESI],
-            [Define to enable the ESI processor and Surrogate header support])
+squid_opt_use_esi=auto
+AH_TEMPLATE([USE_SQUID_ESI],[Define to enable the ESI processor])
 AC_ARG_ENABLE(esi,
-  AS_HELP_STRING([--enable-esi],
-                 [Enable ESI for accelerators. Benefits from expat or libxml2.
+  AS_HELP_STRING([--disable-esi],
```

Comment:
I doubt we should enable ESI where possible because I suspect that most admins do not want ESI processing of responses (and other ESI side effects). At the very least, this change in default should be done/debated separately from this PR.

Reply:
this is catching the documentation up to the change in v3.1 "ESI support default enabled", although there was one case incorrectly still disabled which changes with the library checks being more central to the auto-disable criteria.

Reply:
Ouch! I consider this change request addressed, thank you.

Reply:
However, if the official code was not actually enabling ESI by default in many cases (due to the bug you have mentioned), and we have now "fixed" that, effectively enabling ESI for many unsuspected admins, then we have a bigger mess to deal with.

Reply:
Nod, one of the reasons this is not going near v3.5. If you believe the change is going to be too much of a surprise, I'm fine with leaving it in v5. Though IMO it is still smaller than the other changes already targeted at v4 (eg PR #45 and #81) and the testing these will all need can overlap without obstructing any later fixes.

Reply:
FYI: People using distro packages, or basing their builds on the distro options should be unaffected. I believe the majority of self-builds are just rebuilds of distro packages for SSL-Bump these days, so not many left to surprise.

Reply:
> If you believe the change is going to be too much of a surprise

I do not have enough information to believe either way. This is one of the reasons you should post to squid-users about this change IMHO.